contrib/haproxy/patches/0001-use-env-flags.patch

```diff
@@ -0,0 +1,32 @@
+From da587a4cbd23942459725d70cc785fe0e4bfe160 Mon Sep 17 00:00:00 2001
+From: ttyyls <contact@behri.org>
+Date: Sat, 9 Mar 2024 15:49:18 +0100
+Subject: [PATCH] use env flags
+
+---
+ Makefile | 4 ++--
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/Makefile b/Makefile
+index a66c9b69c..074a27aed 100644
+--- a/Makefile
++++ b/Makefile
+@@ -289,13 +289,13 @@ ARCH_FLAGS        = $(ARCH_FLAGS.$(ARCH))
+ # These CFLAGS contain general optimization options, CPU-specific optimizations
+ # and debug flags. They may be overridden by some distributions which prefer to
+ # set all of them at once instead of playing with the CPU and DEBUG variables.
+-CFLAGS = $(ARCH_FLAGS) $(CPU_CFLAGS) $(DEBUG_CFLAGS) $(SPEC_CFLAGS)
++CFLAGS := $(DEBUG_CFLAGS) $(SPEC_CFLAGS) $(CFLAGS)
+ 
+ #### Common LDFLAGS
+ # These LDFLAGS are used as the first "ld" options, regardless of any library
+ # path or any other option. They may be changed to add any linker-specific
+ # option at the beginning of the ld command line.
+-LDFLAGS = $(ARCH_FLAGS) -g
++LDFLAGS := $(CFLAGS) $(LDFLAGS)
```

Comment:
as far as i can tell, it's not using that (add `V=1` to `make_build_args` and see, while at it leave it there as it's much more useful in logs)